src/Security/Authentication/OAuth/src/OAuthConstants.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Authentication.OAuth
+{
+    /// <summary>
+    /// Constants used in the OAuth protocol
+    /// </summary>
+    public static class OAuthConstants
+    {
+        /// <summary>
+        /// code_verifier defined in https://tools.ietf.org/html/rfc7636
+        /// </summary>
+        public static readonly string CodeVerifierKey = "code_verifier";
```

Comment:
Nit: should these be const, they cannot ever change right?

Reply:
As a rule we only use public const where it's required for use in attributes.